src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java

```diff
@@ -311,6 +314,69 @@ public void testWrongSingletonTagInJavadoc() throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(NonTightHtmlTagIntolerantCheck.class);
+        checkConfig.addAttribute("violateExecutionOnNonTightHtml", "true");
+        final String[] expected = {
+            "6: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "13: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "16: " + getCheckMessage(MSG_UNCLOSED_HTML, "li"),
+            "21: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "27: " + getCheckMessage(MSG_UNCLOSED_HTML, "tr"),
+            "34: " + getCheckMessage(MSG_UNCLOSED_HTML, "li"),
+            "54: " + getCheckMessage(MSG_UNCLOSED_HTML, "li"),
+            "71: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "80: " + getCheckMessage(MSG_UNCLOSED_HTML, "tr"),
+            "124: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+        };
+        verify(checkConfig, getPath("InputAbstractJavadocNonTightHtmlTags.java"), expected);
+    }
+
+    @Test
+    public void testNonTightHtmlTagIntolerantCheckReportingNoViolation() throws Exception {
+        NonTightHtmlTagIntolerantCheck.visitJavadocTokenMethodCallCount = 0;
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(NonTightHtmlTagIntolerantCheck.class);
+        final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
+        verify(checkConfig, getPath("InputAbstractJavadocNonTightHtmlTags.java"), expected);
+    }
+
+    @Test
+    public void testNonTightHtmlTagIntolerantCheckVisitCount()
+            throws Exception {
+        NonTightHtmlTagIntolerantCheck.visitJavadocTokenMethodCallCount = 0;
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(NonTightHtmlTagIntolerantCheck.class);
+        final File testFile = new File(getPath("InputAbstractJavadocNonTightHtmlTags.java"));
+        final List<File> fileList = new ArrayList<>();
+        fileList.add(testFile);
+        final Checker checker = createChecker(checkConfig);
+        checker.process(fileList);
+        assertEquals("Please ensure that method 'visitJavadocToken' is not called "
+                        + "more than thrice.",
+                5, NonTightHtmlTagIntolerantCheck.visitJavadocTokenMethodCallCount);
+    }
+
+    @Test
+    public void testVisitCountForCheckAcceptingJavadocWithNonTightHtml()
+            throws Exception {
+        NonTightHtmlTagTolerantCheck.visitJavadocTokenMethodCallCount = 0;
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(NonTightHtmlTagTolerantCheck.class);
+        checkConfig.addAttribute("violateExecutionOnNonTightHtml", "true");
+        final File testFile = new File(getPath("InputAbstractJavadocNonTightHtmlTags2.java"));
+        final List<File> fileList = new ArrayList<>();
+        fileList.add(testFile);
+        final Checker checker = createChecker(checkConfig);
+        checker.process(fileList);
```

Comment:
Why can't this be a `verify()` method call? Why do we have to manually do all this?

Reply:
@rnveach 
I asked @ps-sp to have only one assert per UT to make them as independent as possible. Since verify `verify()` is meant to assert smth we cannot use it in this case.

Reply:
You should be asserting that the check runs like normal and prints no violations as check has log calls, which is what `verify` is for.
Making multiple new lines of code that we have to maintain just to avoid a simple assertion doesn't make sense to me.

Reply:
OK, my bad, I didn't take into account that specific checks are used and they only count calls and don't log violations.
@ps-sp, please use verify.

Reply:
wait, `AbstractJavadocCheck` can log violations and it is a parent of the used check, so I still think we should not mix assertions:
`testNonTightHtmlTagIntolerantCheckReportingNoViolation` - to assert no violations
`testVisitCountForCheckAcceptingJavadocWithNonTightHtml` - to assert method calls number

@romani, what is your position?

Reply:
> to assert no violations

yes, but Check should raise violation on any call of visitToken , that will allow to use general `verify` method. counting of calls is not preferable.

> to assert method calls number

to assert violations from Checks, not a numbers.

Reply:
@romani Please address the following two things.

1) What `message key` should we use to log violation for `test checks` that are not designed to log violations ? We of course can not create a message key for testing purposed. Say we design the `test check` to actually check something and log violations for it or we make javadoc wrong so that we can somehow use an exisiting key to log violations just so that we can indirectly assert method call count, wouldn't this make the test misleading  ?

2) We want to verify calls to `visitJavadocToken`. If we do that indirectly by using `verify` we mix violations with method call count. Would that be good ?

Reply:
@romani After discussion with @Vladlis we can use mockito or we can create `verify` methods in `...TestSupport` to verify calls to `visitJavadocToken`. Indirectly testing number of times a method has been called by forcefully making it to log some violation and then counting those violations seems to be misleading and to log a violatin we need an appropriate message key and one should not be created just for testing, please correct me if I am wrong on any of the two points. Also, there are other tests which verify call counts to methods in the same way as this test method does. 

@rnveach What's your opinion on this ?

Reply:
@ps-sp 
as discussed with @romani: lets just used a general exception message key
https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/com/puppycrawl/tools/checkstyle/messages.properties#L4

Reply:
@rnveach @romani  @Vladlis  
That property is not accessible to `AbstractJavadocCheck`. We have a version of `getCheckMessage` at [AbstractModuleTestSupport.java#L389](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java#L389) which allows one to specify a different class to be used by `getMessageBundle` but `log` method has no such overloads, **which looks right  implementation to me**.

We can use the version of `getCheckMessageAbove` and call it the following way in `AbstractJavadocCheckTest`
```
getCheckMessage(Class.forName("com.puppycrawl.tools.checkstyle.Checker"),"general.exception")
```
to generate messages like:
```
Expected : .....InputAbstractJavadocNonTightHtmlTags.java:41:8: Got an exception - {0}
```

So we might want to copy key `general.exception` to `message.properties` in javadoc section or do something else.

Meanwhile,

`testBadFile` test at `FileTabCharacterCheckTest` overcomes this using:
```
        final LocalizedMessage localizedMessage = new LocalizedMessage(0,
                Definitions.CHECKSTYLE_BUNDLE, "general.exception",
                new String[] {path + exceptionMessage}, null, getClass(), null);
```
So I will do the same

Reply:
@romani @rnveach 
I propose to stay with current test in this PR and discuss this problem in details in a separate PR or issue

Reply:
@ps-sp , please use any accessible message key, you do not really care about content of message.

Reply:
While rewriting this test I noticed:
```
        final String[] expected = {
            "41:8: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
            "62:13: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
            "99:8: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
            "105:8: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
            "109:8: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
        };
``` 

and 

```
        final String[] expected = {
            "41:8: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
            "105:8: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
            "109:8: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
            "62:13: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
            "99:8: " + getCheckMessage(MSG_SUMMARY_FIRST_SENTENCE),
        };
```

both make the  UT pass. Ordering of expected messages doesn't seem to influence anything. Though I didn't look deeper into this but still. to me it looks like there's something wrong. Is this intended ?

Reply:
Done.

Reply:
@ps-sp 
>  Is this intended ?

Yes. We use a tool the encourages you to write code that doesn't depend on order when implementations don't specify an order. The tool is called `NonDex`.
Because of this we automatically sort things, like expected UT results. Sorting happens after run of Checkstyle, so it is only on the UT side and not in production code.

Reply:
@rnveach 
Thank you.